### ISSUE TITLE
[Enhancement] Difficulty modifier

### ISSUE THREAD

**Issue by jafdy** [endless-sky/endless-sky#1386]
http://steamcommunity.com/app/404410/discussions/0/360670708786325536/?tscn=1469809906

For those people who think the game is too easy.

**Comment by KiLEdEnNis** [comment]
the only ways I can see to do this is either making a harder difficulty that expects you to be a certain strength, or that enemies' numbers SCALE on your ships

but I do like the idea as I played the main story like 8x (I have a life I swear) and whenever I start I get an interceptor, plunder/hijack and have a heavy warship before the memorial day

although, wasn't there a plan for hardcore/iron man/whatever mode where dead means dead without reloading?

**Comment by MarcelineVQ** [comment]
I've got a bit of that in the works at https://github.com/MarcelineVQ/endless-sky/wiki/Permadeath-and-Hardcore but it's just the very basics at the moment. Ideally there'd be all sorts of mode toggles like turning off the ability to buy or sell ships or outfits.
It's not really different difficulties though, more like challenges for the player.

**Comment by Amazinite** [comment]
#362 for reference to an earlier perma death idea.

**Comment by jafdy** [comment]
I think he wanted a toggle or a slider that would increase the size of mission fleets.

**Comment by Alkallid** [comment]
That would need lots of changes and probably fuck up mission creating.

**Comment by KiLEdEnNis** [comment]
maybe a modifier that makes enemy fleets more common and allied fleets less common, but that may mess the game up as well, and you can't make a mode where enemy ships have more shields and damage

**Comment by endless-sky** [comment]
It's like any RPG - if you allow the player to play the same "area" over and over again, they can choose to accumulate enough XP that they will be able to beat the rest of the game easily. That is, when people say it's "too easy" what they really mean is that they spent hours "leveling up" before starting the story.

**Comment by KiLEdEnNis** [comment]
that's true but an experienced player can get a heavy wartship + escorts before the memorial event (by hijacking)
but maybe the player can make it more challenging by keeping only 1 ship

**Comment by Darklocq** [comment]
Was going to suggest this in a new ticket, then found this one.  So, pasting it here:

While I'm overjoyed to find Endless Sky (I was really missing Escape Velocity, and have tried several homages to it, and this is the best by far), I'm finding it too easy.  This could use a difficulty slider that jacked up (or optionally reduced) various in-game variables with some fractional multiplier.  On a scale of 1–10 (easy–hard), I'd say the current default game mechanics are perhaps a 4.

In my case, I was able to get to over 2.5M credits (and loan creditworthiness for another 1.2M) in a single afternoon–evening gaming session, still flying my original shuttle (with upgrades), ignoring asteroid mining and most commodities trading, and doing pretty much nothing but the tutorial missions, some minor side quest, and a boatload of cargo and passenger missions (and one smuggling incident), with no combat other than fleeing or occasionally dying (I'm running with no weapons, just speed, until I can afford a good middle-weight ship).  I even lost 600K credits due to some badly implemented trade-route thing in the "Businessman" mod, and still have enough on hand and in credit to buy a 3.7M ship. My death rate is much less than what I expected, and most often caused by not paying attention right after I jump into a system; I probably could have avoided 50% of these.  I'm dying about 1/4 as often as I used to in EVN.  Various main-story events have started (wars, terrorist attacks), but I'm outside of most of them so far due to no combat rating.

Some key things to change with a difficulty slider:
* Amount of pay for jobs (at 100% difficulty it should probably be about 1/3 what the default is)
* Number of jobs that are time-constrained (maybe up to double).
* Number of jobs available (maybe down to half)
* Frequency of pirate attacks (up to perhaps 3x) and maybe also of patrols, in systems they control (up to 2x?)
* Number of ships in pirate and patrol fleets (up to at least 2x)
* Average distance away of jobs (up to 1.5x maybe - enough that many rush jobs cannot be accepted without serious hyperdrive and fuel upgrades)
* Possibly also an increase in weapons and other gear on NPC ships, though that could introduce a balance problem of its own, if you're good enough to board them with any frequency.
* An alternative would be what the Elder Scrolls games do with their difficulty slider: instead of different gear and stats, a multiplier "filter" is applied, such that your own attacks do less damage to enemies, and their attacks (against player only, not each other) do more.
* Various things can also be made relative to player character experience as reflected in various stats – with or without a difficulty slider.  E.g., jack up the number and difficulty of pirates if you have a high combat rating, or lots of credits, but not too much if you are in a crappy ship.  I find it weird that I'm basically a semi-rich courier with a reputation (by now, I would think) of quickly running multiple jobs, at max cargo and passenger capacity, and with about a 98% success rate (I've only failed 2 rush jobs), yet I don't attract more frequent attention from pirates.  Nor does it seem to matter what I'm hauling. Nor whether I'm hauling it from one faction's jurisdiction to another.
* Commodity-trading values could actually go UP a bit, to encourage if not force at least some reliance on hauling around stuff like Heavy Metals and Medical supplies.

I have to disagree with this comment, at least in this context:

> It's like any RPG - if you allow the player to play the same "area" over and over again, they can choose to accumulate enough XP that they will be able to beat the rest of the game easily. That is, when people say it's "too easy" what they really mean is that they spent hours "leveling up" before starting the story.

But this is an open-ended game by design, with no specific, pre-determined way to start or to progress. If you want to be a fighter jock right out of the gate, you can. If you want to be a transport courier, that's up to you. Big into asteroid mining? Then have at it.  If your goal is tight integration into a faction and eventually captaining their largest war fleet, go for it.  It's not a "mistake" or an "exploit" to build up sufficient credit to buy a combat-worthy vessel instead of going around getting blown up 15 times an hour. (Analogy: if you're going to be a boxer, you train hard and bodybuild and work up a pain tolerance and a here-to-win spirit, you don't just wake up one day, decide you want to box, and go enter a professional bout, or you'll end up in the hospital.) 

If the devs want to make it mandatory to focus on a mixture of fighting and transport and politicking from the start, then they'll have to make sure every single ship comes with at least one weapon (and can equip a non-trivial one) AND with sufficient minimum cargo and bunk space (8, and 2 passengers?) for some routine jobs. Then not permit anyone to sell off their last remaining weapon, nor convert (even with mods) 100% of cargo and passenger space to something else. Otherwise, people are gonna do exactly what they want, which is what they expect from such a game in the first place.  And the game would likely need to reduce the combat-reputation requirements for various quest-launching tests.  I'm basically running a high-end courier service in my game's early phase (the Millennium Falcon, but even faster and with no guns, ha ha).  This isn't "wrong" or "cheating" or "grinding"; it's just my game.

The EV games were never "too easy" by virtue of focusing on any particular aspect like transport or boarding (though every game in this vein seems to short-circuit its own commodities-trading mechanics by making other means of gaining credits both less hassle and more lucrative, even from early in the game).  Even when EVN, for example, got kind of easy (after you built up a reasonable merchant–mercenary fleet) if you did nothing but go after pirates and profit off them, it was still hard in other ways, like all-out warfare, and then recovering from it (e.g. doing enough diplomacy to get back in good enough graces with former enemies to complete quests in their territory).

Anyway, I don't want any of this to sound like a complaint; it's just ideas for balance options and development, and a gaming-philosophy demurrer.  I'm amazed at the quality of the game, and that it's free, and has a bit of a volunteer dev community working on it, too.  I'm not even actually bummed that getting to the 3mil credits mark hasn't been that hard, since that's basically just the beginning of being a contender, able to get a ship that doesn't asplode within 30 seconds of getting in a fight.  Still, I would have expected that part to take half a week, not one day.  :-)
